gsy_framework/sim_results/carbon_emissions/results.py

```diff
@@ -42,31 +42,40 @@ def _find_start_and_end_dates(self, dates: List) -> Tuple[DateTime, DateTime]:
 
     def _create_hourly_timestamps(self, start: DateTime, end: DateTime) -> List[DateTime]:
         """Generate a list of datetime objects for each hour between start and end"""
+        if not isinstance(start, pendulum.DateTime):
+            start = pendulum.instance(start)
+        if not isinstance(end, pendulum.DateTime):
+            end = pendulum.instance(end)
         return [start.add(hours=hour) for hour in range(int((end - start).total_hours()) + 1)]
 
     def query_carbon_ratio_from_static_source(
-        self, country_code: str, start: DateTime, end: DateTime
+        self, country_code: str, start: DateTime, end: DateTime, file_prefix=""
     ) -> List[Dict]:
         """Calculate the total carbon generated based on the specified statistic"""
 
+        if not file_prefix:
+            file_prefix = "gsy_framework"
+
         if start.tzinfo is None or end.tzinfo is None:
             raise ValueError("start and end must have timezone")
-        if start.timezone_name != TIME_ZONE or end.timezone_name != TIME_ZONE:
+        if start.astimezone().tzname() != TIME_ZONE or end.astimezone().tzname() != TIME_ZONE:
```

Comment:
why was this needed? Arend the `start` and `end` pendulum times?

Reply:
Let me check if this is no longer needed. It was crashing before I converted the start / end time from normal datetimes (received from GraphQL / graphene) to penulum datetimes, but I have meanwhile converted those to pendulum

Reply:
It is required because this clause used to be wrong in the first place. TIME_ZONE is a string, and start / end are DateTime objects, therefore we really need the timezone string of the datetime for comparison.

Reply:
Yes, and start.timezone_name returns a string which should have worked also in the original version.